### PR TITLE
Use black for formating library and add initial testing (will fail)

### PR DESCRIPTION
**Pull request by jorgensd**
- Use black to format library. There are still alot of commented code that will not pass a `python3 -m flake8` run, but it is at least a start.
- I've removed a lot of unused imports.
- Various refactoring of imports, using relative imports inside module and avoid infinite import loop
- Point to more up to date Docker image of DOLFIN, supplied my me and @finsberg  in https://github.com/scientificcomputing/packages
- Create initial CI running the tests of Stubs. note that these does not pass. I tried to modify the test such that it would, but Im getting an assertion error:
```bash
=========================================================================================================================================================== FAILURES ============================================================================================================================================================
_____________________________________________________________________________________________________________________________________________________ test_stubs_model_init _____________________________________________________________________________________________________________________________________________________

stubs_model = Model(pc=<stubs.model_assembly.ParameterContainer object at 0x7ff0a2d07208>, sc=<stubs.model_assembly.SpeciesContainer...=<stubs.config.Config object at 0x7ff0a2d07ef0>, parent_mesh=<stubs.mesh.ParentMesh object at 0x7ff0a2d07320>, name='')

    @pytest.mark.stubs_model_init
    def test_stubs_model_init(stubs_model):
        "Test the different parts of model initialization"
        # initialize
        model = stubs_model
    
        model._init_1()
>       model._init_2()

test_stubs_model_init.py:88: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../stubs/model.py:281: in _init_2
    self._init_2_3_link_reaction_properties()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = Model(pc=<stubs.model_assembly.ParameterContainer object at 0x7ff0a2d07208>, sc=<stubs.model_assembly.SpeciesContainer...=<stubs.config.Config object at 0x7ff0a2d07ef0>, parent_mesh=<stubs.mesh.ParentMesh object at 0x7ff0a2d07320>, name='')

    def _init_2_3_link_reaction_properties(self):
        fancy_print(
            f"Linking parameters, species, and compartments to reactions",
            format_type="log",
        )
    
        for reaction in self.rc:
            reaction.parameters = {
                param_name: self.pc[param_name]
                for param_name in reaction.param_map.values()
            }
            reaction.species = {
                species_name: self.sc[species_name]
                for species_name in reaction.species_map.values()
            }
            compartment_names = [
                species.compartment_name for species in reaction.species.values()
            ]
            if reaction.explicit_restriction_to_domain:
                compartment_names.append(reaction.explicit_restriction_to_domain)
            reaction.compartments = {
                compartment_name: self.cc[compartment_name]
                for compartment_name in compartment_names
            }
            # number of parameters, species, and compartments
            reaction.num_parmeters = len(reaction.parameters)
            reaction.num_species = len(reaction.species)
            reaction.num_compartments = len(reaction.compartments)
    
            is_volume = [c.is_volume for c in reaction.compartments.values()]
            if len(is_volume) == 1:
                if is_volume[0]:
                    reaction.topology = "volume"
                else:
                    reaction.topology = "surface"
            elif len(is_volume) == 2:
                if all(is_volume):
                    raise NotImplementedError(
>                       f"Reaction {reaction.name} has two volumes - the adjoining surface must be specified "
                        f"using reaction.explicit_restriction_to_domain"
                    )
E                   NotImplementedError: Reaction A <-> A_er has two volumes - the adjoining surface must be specified using reaction.explicit_restriction_to_domain

../stubs/model.py:506: NotImplementedError
============================================================================================================================================== 1 failed, 2 passed in 0.48 seconds ===============================================================================================================================================
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,10 +16,10 @@ Systems biology models are converted by STUBS into the appropriate systems of re
 Although FEniCS is a core dependency, because it has many different versions (2019.1, development, FEniCSx, etc.), is quite large, and is complicated to build, it is not packaged with STUBS by default. The recommended way to use STUBS is to create a container from one of the official FEniCS docker images and to pip install STUBS from within the container.
 
 ```bash
-# create a container from the official fenics docker image
-jgl:~$ docker run -ti --init quay.io/fenicsproject/dev
+# create a container using DOLFIN built on ubuntu 22.04 with Python 3.10
+jgl:~$ docker run -ti --init ghcr.io/scientificcomputing/fenics:2022-12-12
 # pip install stubs from within the container
-fenics:~$ pip install stubs
+root@jgl:~$ python3 -m pip install stubs
```
Comment (finsberg) [inline README.md]: Is this correct? It does not look like the same repo https://pypi.org/project/stubs/

Should it be `fenics-stubs`: https://pypi.org/project/fenics-stubs/ ?
We should also make sure to set up workflow that pushes out a new version to pypi on every new release.

Comment (jorgensd) [reply README.md]: 👍🏻

Comment (jorgensd) [inline README.md]:
```suggestion
root@jgl:~$ python3 -m pip install fenics-stubs
```

**Review by justinlaughlin — APPROVED** [review]
This looks great - thank you!

**Pull request merged**